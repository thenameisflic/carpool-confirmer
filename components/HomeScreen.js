import React from "react";
import { TouchableOpacity, View, StyleSheet, Image } from "react-native";
import Title from "./typography/Title";
import theme from "../theme";

export default function Home({navigation}) {
  const { navigate } = navigation;

  return (
    <View style={styles.container}>
      <Title size="h4" style={styles.title}>Selecione o seu perfil</Title>
      <View style={styles.row}>
        <TouchableOpacity style={styles.role} onPress={() => navigate("Driver")}>
          <Image style={styles.roleImage} source={{uri: "https://placehold.it/120x120?text=Motorista"}} />
          <Title style={styles.roleTitle} size="h6">Motorista</Title>
        </TouchableOpacity>
        <TouchableOpacity style={styles.role} onPress={() => navigate("Passenger")}>
          <Image style={styles.roleImage} source={{uri: "https://placehold.it/120x120?text=Passageiro"}} />
          <Title style={styles.roleTitle} size="h6">Passageiro</Title>
        </TouchableOpacity>
      </View>
    </View>
  );
}

Home.navigationOptions = {
  header: null
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "stretch",
    backgroundColor: theme.backgroundColor,
    paddingHorizontal: 8
  },
  row: {
    flexDirection: "row",
    justifyContent: "space-around",
    marginTop: 40
  },
  title: {
    textAlign: "center"
  },
  role: {
    alignItems: "center"
  },
  roleTitle: {
    marginTop: 8,
    color: theme.primaryColor
  },
  roleImage: {
    width: 120,
    height: 120,
    borderRadius: 60
  }
});
