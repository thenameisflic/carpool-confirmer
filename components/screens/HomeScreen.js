import React, { useEffect } from "react";
import { TouchableOpacity, View, StyleSheet, Image } from "react-native";
import Title from "../typography/Title";
import theme from "../../theme";
import DriverIcon from "../../assets/images/DriverImg.png";
import PassengerIcon from "../../assets/images/PassengerImg.png";
import RNLocation from "react-native-location";

export default function Home({navigation}) {
  const { navigate } = navigation;

  useEffect(() => {
    RNLocation.requestPermission({
      ios: "whenInUse",
      android: {
        detail: "coarse",
        rationale: {
          title: "Nós precisamos de acesso à sua localização",
          message: "",
          buttonPositive: "OK",
          buttonNegative: "Cancelar"
        }
      }
    });
    RNLocation.configure({ distanceFilter: 0 });
  }, []);

  return (
    <View style={styles.container}>
      <Title size="h4" style={styles.title}>Selecione o seu perfil</Title>
      <View style={styles.row}>
        <TouchableOpacity style={styles.role} onPress={() => navigate("Driver")}>
          <Image style={styles.roleImage} source={DriverIcon} />
          <Title style={styles.roleTitle} size="h6">Motorista</Title>
        </TouchableOpacity>
        <TouchableOpacity style={styles.role} onPress={() => navigate("Passenger")}>
          <Image style={styles.roleImage} source={PassengerIcon} />
          <Title style={styles.roleTitle} size="h6">Passageiro</Title>
        </TouchableOpacity>
      </View>
    </View>
  );
}

Home.navigationOptions = {
  header: null
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "stretch",
    backgroundColor: theme.backgroundColor,
    paddingHorizontal: 8
  },
  row: {
    flexDirection: "row",
    justifyContent: "space-around",
    marginTop: 40
  },
  title: {
    textAlign: "center"
  },
  role: {
    alignItems: "center"
  },
  roleTitle: {
    marginTop: 8,
    color: theme.primaryColor
  },
  roleImage: {
    width: 120,
    height: 120,
    borderRadius: 60
  }
});
