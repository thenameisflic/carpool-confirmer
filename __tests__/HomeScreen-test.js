import "react-native";
import React from "react";
import HomeScreen from "../components/screens/HomeScreen";
import { TouchableOpacity } from "react-native";

import renderer from "react-test-renderer";

jest.mock("react-native-location", () => ({requestPermission: () => {}, configure: () => {}}))

it("renders HomeScreen correctly", () => {
  const tree = renderer.create(<HomeScreen navigation={{navigate: ""}} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("navigates to correct screen when clicking on a TouchableOpacity", () => {
  const navigateMock = jest.fn();
  const instance = renderer.create(<HomeScreen navigation={{navigate: navigateMock}} />);
  const driverBtn = instance.root.findAllByType(TouchableOpacity)[0];
  const passengerBtn = instance.root.findAllByType(TouchableOpacity)[1];
  driverBtn.props.onPress();
  passengerBtn.props.onPress();
  expect(navigateMock).toBeCalledWith(expect.stringMatching("Driver"));
  expect(navigateMock).toBeCalledWith(expect.stringMatching("Passenger"));
});
