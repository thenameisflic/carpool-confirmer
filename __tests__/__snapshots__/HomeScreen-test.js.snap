// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders HomeScreen correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "Nunito Regular",
          "fontSize": 34,
        },
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    Selecione o seu perfil
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginTop": 40,
      }
    }
  >
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/DriverImg.png",
          }
        }
        style={
          Object {
            "borderRadius": 60,
            "height": 120,
            "width": 120,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito Bold",
              "fontSize": 20,
            },
            Object {
              "color": "#FF5E57",
              "marginTop": 8,
            },
          ]
        }
      >
        Motorista
      </Text>
    </View>
    <View
      accessible={true}
      clickable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/PassengerImg.png",
          }
        }
        style={
          Object {
            "borderRadius": 60,
            "height": 120,
            "width": 120,
          }
        }
      />
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito Bold",
              "fontSize": 20,
            },
            Object {
              "color": "#FF5E57",
              "marginTop": 8,
            },
          ]
        }
      >
        Passageiro
      </Text>
    </View>
  </View>
</View>
`;
